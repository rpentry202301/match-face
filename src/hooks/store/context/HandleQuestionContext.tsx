"use client";

import React, { useContext } from "react";

type Refine = {
  word: string[];
  department: number[];
};

const SelectContext = React.createContext<[Refine, React.Dispatch<React.SetStateAction<Refine>>] | undefined>(undefined);

export function SelectProvider({ children }: { children: React.ReactNode }) {
  const [select, setSelect] = React.useState<Refine>({
    word: [],
    department: [],
  });

  return (
    <SelectContext.Provider value={[select, setSelect]}>
      {children}
    </SelectContext.Provider>
  );
}

export function useRefine() {
  const context = useContext(SelectContext);
  if (context === undefined) {
    throw new Error("useRefine must be used within a SelectProvider");
  }
  return context;
}
