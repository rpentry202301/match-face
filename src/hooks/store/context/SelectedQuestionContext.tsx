"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { QusetionsReducer } from "../reducer/selecterReducer";
import { QuestionsReducerAction } from "@/types/admin/tasks/register/types";

export type Questions = {
  projectId: number;
  list: {
    id: number;
    name: string;
  }[];
};

const SelectedQuestionContext = createContext<
  [Questions, Dispatch<QuestionsReducerAction>] | undefined
>(undefined);

export const SelectedQuestionProvider = ({
  children,
}: {
  children: ReactNode;
}) => {
  const [state, dispatch] = useReducer<
    Reducer<Questions, QuestionsReducerAction>
  >(QusetionsReducer, { projectId: 0, list: [] });
  return (
    <SelectedQuestionContext.Provider value={[state, dispatch]}>
      {children}
    </SelectedQuestionContext.Provider>
  );
};

export const useSelectedQuestion = () => {
  const context = useContext(SelectedQuestionContext);
  if (context === undefined)
    throw new Error("Function 'useSelectedQuestion' must be used in Provider.");
  return context;
};
