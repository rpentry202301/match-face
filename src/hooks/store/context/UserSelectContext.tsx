"use client";
import {
  Dispatch,
  ReactNode,
  Reducer,
  createContext,
  useContext,
  useReducer,
} from "react";
import { UsersReducer } from "../reducer/selecterReducer";
import { UsersReducerAction } from "@/types/admin/tasks/register/types";

export type Users = {
  id: string;
  name: string;
};

const UserSelectContext = createContext<
  [Users[], Dispatch<UsersReducerAction>] | undefined
>(undefined);

export const UserSelectProvider = ({ children }: { children: ReactNode }) => {
  const [state, dispatch] = useReducer<Reducer<Users[], UsersReducerAction>>(
    // selecterReducer,
    UsersReducer,
    []
  );
  return (
    <UserSelectContext.Provider value={[state, dispatch]}>
      {children}
    </UserSelectContext.Provider>
  );
};

export const useUserSelect = () => {
  const context = useContext(UserSelectContext);
  if (context === undefined)
    throw new Error("Function 'useUserSelect' must be used in Provider.");
  return context;
};
