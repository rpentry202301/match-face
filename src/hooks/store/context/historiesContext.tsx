'use client'
import { createContext, Dispatch, useContext, useReducer } from "react";
import { SelectHistoryAction } from "@/types/admin/histories/admin_histories";
import { SelectHistoryReducer } from "../reducer/historiesReducer";

const SelectHistoryContext = createContext<{month:string,department:number[],skills:number[]}| undefined>(undefined);
const SelectHistoryDispatchContext = createContext<Dispatch<SelectHistoryAction>|undefined>(undefined)
const initialSelect = {month:"",department:[],skills:[]}

export function SelectHistoryProvider ({children}:{children:React.ReactNode}){
    const [state,dispatch] = useReducer(SelectHistoryReducer,initialSelect)
    return (
        <SelectHistoryContext.Provider value={state}>
            <SelectHistoryDispatchContext.Provider value={dispatch} >
            {children}
            </SelectHistoryDispatchContext.Provider>
        </SelectHistoryContext.Provider>
    )
}

export const useSelectHistory = () => {
    const context = useContext(SelectHistoryContext)
    if(context === undefined) {
        throw new Error("Function 'useSelectHistory' must be used in Provider.")
    }
    return context
}

export const useSelectHistoryDispatch = () => {
    const dispatch = useContext(SelectHistoryDispatchContext)
    if(dispatch === undefined) {
        throw new Error("Function 'useSelectHistoryDispatch' must be used in Provider.")
    }
    return dispatch
}
