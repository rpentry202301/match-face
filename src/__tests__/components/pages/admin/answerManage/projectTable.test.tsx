import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import ProjectTable from "@/components/pages/admin/answerManage/projectTable";
import React from "react";
import { SelectProvider } from "@/hooks/store/context/HandleQuestionContext";
import "@testing-library/jest-dom";

type Project = {
  createdAt: string;
  createdUser: string;
  deleted: boolean;
  departmentId: number;
  detail: string;
  enterpriseId: number;
  id: number;
  name: string;
  questionList: any[];
  updateAt: string;
  updateUser: string;
};

const mockData = [
  {
    id: 4,
    name: "test案件",
    detail: "販促アプリの新規開発、既存システムの保守・運用。",
    enterpriseId: 0,
    departmentId: 0,
    questionList: [],
    createdUser: "テスト花子",
    createdAt: "2023-09-04T16:59:11.203322",
    updateUser: "テスト花子",
    updateAt: "2023-09-04T16:59:11.203322",
    deleted: false,
  },
  {
    id: 1,
    name: "バックエンド案件",
    detail: "販促アプリの新規開発、既存システムの保守・運用。",
    enterpriseId: 0,
    departmentId: 0,
    questionList: [],
    createdUser: "テスト花子",
    createdAt: "2023-09-04T10:35:01.831993",
    updateUser: "テスト花子",
    updateAt: "2023-09-04T10:35:01.831993",
    deleted: false,
  },
];

describe("質問・回答例一覧画面テスト", () => {
  beforeEach(() => {
    const mockFetchPromise = Promise.resolve({
      json: () => Promise.resolve(mockData),
    });

    global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("スナップショット", async () => {
    const { container } = render(
      <SelectProvider>
        <ProjectTable />
      </SelectProvider>
    );
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
