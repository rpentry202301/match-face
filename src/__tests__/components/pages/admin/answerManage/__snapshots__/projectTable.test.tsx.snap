// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`質問・回答例一覧画面テスト スナップショット 1`] = `
<div>
  <table
    class="w-4/5 border-2"
    data-testid="projectTable"
  >
    <tbody>
      <tr
        class="border-2 border-current bg-light-gray"
      >
        <th
          class="w-1/5 border-2 py-4"
        >
          最終編集日
        </th>
        <th
          class="w-1/5 border-2"
        >
          案件名
        </th>
        <th
          class="w-2/5 border-2"
        >
          案件概要
        </th>
        <th
          class="w-1/5 border-2"
        />
      </tr>
    </tbody>
  </table>
  <div
    class="w-5/12 max-w-5/12 flex justify-center my-7"
  >
    <ul
      class="flex"
    />
  </div>
</div>
`;
